x-pack/legacy/plugins/infra/index.ts

```diff
@@ -71,7 +74,27 @@ export function infra(kibana: any) {
       return getConfigSchema(Joi);
     },
     init(server: any) {
-      initServerWithKibana(server);
+      // convert hapi instance to KbnServer
+      // `kbnServer.server` is the same hapi instance
+      // `kbnServer.newPlatform` has important values
+      const kbnServer = (server as unknown) as KbnServer;
+      const { core } = kbnServer.newPlatform.setup;
+
+      const getConfig$ = <T>() =>
+        new Observable<T>(observer => {
+          observer.next(server.config().get('xpack.infra'));
+        });
+
+      const initContext = {
+        config: {
+          create: getConfig$,
+          createIfExists: getConfig$,
+        },
+      } as PluginInitializerContext;
```

Comment:
This is a small shim for the config service observable layer. Definitely need some checks to see if this is correctly implemented and then maybe it could be moved somewhere central for others to use during migration.